tests/RecipeForm.test.js

```diff
@@ -358,4 +360,84 @@ describe("The `RecipeForm` component", () => {
       updatedRecipeData,
     );
   });
+
+  it("save button is re-enabled when form errors are addressed", async () => {
+    const recipeData = setup();
+    const { getByText, getAllByRole } = await render(<App />);
+
+    fireEvent.click(getByText("Recipes"));
+
+    await waitFor(() => {
+      expect(getByText("Edit Recipe")).toBeInTheDocument();
+    });
+    fireEvent.click(getByText("Edit Recipe"));
+    await waitFor(() => {
+      expect(getByText("Experimenter Slug")).toBeInTheDocument();
+    });
+
+    const formGroups = getAllByRole("group");
+    const { fallbackFOForm } = getForms(formGroups);
+
+    const saveButton = getByText("Save");
+    expect(saveButton).not.toHaveAttribute("disabled");
+
+    const foCodeBlock = fallbackFOForm.querySelector("textarea");
+
+    const clipboardEvent = new Event("paste", {
+      bubbles: true,
+      cancelable: true,
+      composed: true,
+    });
+
+    clipboardEvent.clipboardData = {
+      getData: () => "invalid json!",
+    };
+    foCodeBlock.dispatchEvent(clipboardEvent);
+
+    expect(getByText("[]invalid json!")).toBeInTheDocument();
+    expect(getByText("Filter Object(s) is not valid JSON")).toBeInTheDocument();
+    expect(
+      getByText("Filter Object(s) is not contained in an array"),
+    ).toBeInTheDocument();
+    expect(saveButton).toHaveAttribute("disabled");
+
+    //ctrl z... the paste
+    foCodeBlock.dispatchEvent(new Event("focus"));
+    foCodeBlock.dispatchEvent(
+      new KeyboardEvent("keydown", { keyCode: 90, ctrlKey: true }),
+    );
+
```

Comment:
A kind of round about way to change the value generic FO....
I wasn't able to use`fireEvent`to change it, nor was I able to use an Input Event to change it either 😬